enhancements/scheduling/load-aware-vm-rebalancing-through-descheduler.md

```diff
@@ -0,0 +1,472 @@
+    - Should we impose a limit on the overall rebalancing cost or the number of live migrations per unit of time?
+    - Is the current chaotic nature of rebalancing acceptable from a cost perspective?
+  Uncontrolled rebalancing could lead to excessive migrations, reducing overall efficiency and potentially degrading workload performance. Introducing constraints—such as rate-limiting migrations or evaluating cost thresholds—could help strike a balance between achieving a well-balanced cluster and minimizing unnecessary disruptions.
+ > **Answer**: Currently, there is no SLA defining the expected duration for rebalancing or the maximum number of allowed live migrations. However, an SLA may be introduced in the future based on customer feedback.
+ > 1. **Descheduler overstepping:** Should the descheduler provide scheduling instructions by soft-tainting over-utilized nodes?
+ > **Answer**: While this could be an interesting avenue to explore, there are currently no plans to extend the descheduler with the ability to provide scheduling hints.
+ > 1. **Utilization thresholds:** How to set the thresholds for practical rebalancing? What guidedance/tips to share with administrators to help them set the thresholds right?
+ > **Answer**: The descheduler operator will provide a set of tuning options to help administrators achieve the optimal balance. Additional tuning options or enhancements may be considered based on customer feedback.
+
+## Test Plan
+
+* Unit tests are added to verify the integration of `LowNodeUtilization` with Prometheus metrics.
+* Unit tests are added to validate the functionality of the new taint controller.
+* End-to-end (e2e) tests are added to ensure the taint controller can only add or remove a specific soft taint when updating a node.
+* The integration of the descheduler, taint controller, and a CNV cluster is tested to verify overall functionality during VM rebalancing.
+
+## Graduation Criteria
+
+Currently targeted as a Tech Preview, with graduation to GA dependent on customer feedback from the available experimental customization.
+
+Rebalancing is an inherently chaotic process. Multiple iterations of customer feedback may be necessary to refine the solution, ensuring it performs well and meets the standards required for a GA release.
+
+### Dev Preview -> Tech Preview
+
+- The Descheduler is integrated with Prometheus and can evict pods based on their actual resource utilization
+- The Kubernetes Scheduler is guided to favor overutilized pods less
+- A new descheduler operator profile is provided
+- List of various tuning options for the load-aware workload rebalancing is provided
+
+### Tech Preview -> GA
+
+- List of key tuning options is identified
+- Customers confirm the tuning options lead to sufficiently fast rebalancing
+- The new descheduler operator profile is available by default
```

Comment:
I'd like to mention that we should also provide sane defaults (ideally to fit a relevant amount of typical cluster sizes) for all the profileCustomizations coming with the new profile.

Reply:
Done.